pulp_ansible/app/viewsets.py

```diff
@@ -632,6 +637,24 @@ class AnsibleRepositoryViewSet(RepositoryViewSet, ModifyRepositoryActionMixin, R
         "ansible.ansiblerepository_viewer": ["ansible.view_ansiblerepository"],
     }
 
+    def get_ansible_repository_qs(self, qs, repo_perm):
+        """
+        Returns a queryset by filtering out private repositories unless the user has the required
+        permission to view.
```

Comment:
I just want to check, but galaxy wants the behavior for repositories to be "visible to any authenticated user if it is public, else hidden unless you have view permission"? This differs from Pulp's general policy of "only user with permissions can see the repo".

Reply:
I don't understand, your latest change reverted the behavior back to how the original `scope_queryset` method worked. Your custom method is making no real distinction between private and non-private repos now. What is the actual behavior a user should expect when they set the `private` flag to True?